packages/samples/test/output/rest/petstore/openapi.json

```diff
@@ -1262,14 +1262,30 @@
             "description": "The items on this page"
           },
           "nextLink": {
-            "type": "string",
+            "$ref": "#/components/schemas/Rest.ResourceLocation_Owner",
             "description": "The link to the next page of items"
           }
         },
         "description": "Paged response",
         "required": [
           "value"
         ]
+      },
+      "Rest.ResourceLocation_Pet": {
```

Comment:
This is unfortunate, I don't think we currently have a way to mark a model as pass-through to not make its own schema.  I didn't add a `friendlyName` here yet because I don't think we really want a named schema for this.

Thoughts? @bterlson @johanste

Reply:
With #562, it would get inlined without a friendly name. It sounds like that's what you want here?

Reply:
That's what I'd hope for, but does it work for plain model properties?

Reply:
I don't think I understand. Can you give me an example of the case in question?

Reply:
`ResourceLocation<T>` will be used in cases like this:

```
model Page<T> {
  nextLink?: ResourceLocation<T>
}
```

Will the generic name elision turn `ResourceLocation<T>` into a plain `string` inline for the `Page<T>` schema?

Reply:
I see. I think so, but I'm checking. BRB. :)

Reply:
Yes, should work fine. I found a trick to force inlining as a test of the future behavior. We bail when generic instantiation is nested, so I added a dummy type argument to force nesting and trigger the inline code path. The fix for #562 would just go down this path for every generic instantiation without a friendly name, regardless of nesting.

https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwoKdXNpbmcgQ2FkbC5IdHRwOwoKbW9kZWwgUGV0IHsKICBuYW1lOiBzdHJpbmc7Cn3IH0ZvcmNlSW5saW5pbmc8VCA9IHVua25vd24%2BIHt9CgpAZG9jKCJMb2NhdGlvbiBvZiB7xEl9IiwgVCnHQ1Jlc291cmNlyCY8VCwgXyA9zls%2BIGlzyX7HOVBhZ2U8VD7lAJ9pdGVtczogVFtdO%2BQArWV4dExpbms6010%2B5QC%2Bb3AgbGlzdCgpOsZHUGV0Pjs%3D

Reply:
(Incidentally, we have always bailed in the nested case though it used to be obscured by a regex that wasn't able to cope and is now explicitly handled without any regex crimes. This is one of the main reasons that I felt the rules for inline-or-not were too unpredictable and that a blanket "if you want non-inline, give me a friendly name rule" would be much better, which became #562)

Reply:
> maybe I can just leave these unfortunate schema additions in the output files until they get taken back out by your change.

Sounds fine to me!

Reply:
> Yes, should work fine. I found a trick to force inlining as a test of the future behavior. We bail when generic instantiation is nested, so I added a dummy type argument to force nesting and trigger the inline code path. The fix for #562 would just go down this path for every generic instantiation without a friendly name, regardless of nesting.
> 
> https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwoKdXNpbmcgQ2FkbC5IdHRwOwoKbW9kZWwgUGV0IHsKICBuYW1lOiBzdHJpbmc7Cn3IH0ZvcmNlSW5saW5pbmc8VCA9IHVua25vd24%2BIHt9CgpAZG9jKCJMb2NhdGlvbiBvZiB7xEl9IiwgVCnHQ1Jlc291cmNlyCY8VCwgXyA9zls%2BIGlzyX7HOVBhZ2U8VD7lAJ9pdGVtczogVFtdO%2BQArWV4dExpbms6010%2B5QC%2Bb3AgbGlzdCgpOsZHUGV0Pjs%3D

Is this something we have to do specifically for the openapi family of emitters? If so, it feels a bit unfortunate to have to put in our core Azure API pattern libraries... Remember, the main reason we've had issues with inlined models is codegen giving them names - but we are not going to consume openapi in our code generators, we are going to consume the Cadl directly...

Reply:
To be clear, the trick was only to demonstrate what the future behavior would be after #562. It's a trick to go down the code path we would take once #562 is implemented. And with that this would be inline with no name and trickes, which sounds like what everyone wants or are you asking for something else?

Reply:
We're not checking in the trick, we're waiting for #562 to come along and inline this without it.

Reply:
Ok - that was my only concern. Thanks!

Reply:
I went ahead and fixed #562

PRs: 

https://github.com/microsoft/cadl/pull/796
https://github.com/Azure/cadl-azure/pull/1782